enhancements/monitoring/replace-prometheus-adapter-with-metrics-server.md

```diff
@@ -0,0 +1,294 @@
+  }
+  ```
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Configuration/Enablement
+
+Metrics Server is not enabled by default. It must be explicitly enabled by the user. It is enabled by setting 
+`enabled: true` in the ConfigMap `openshift-monitoring/cluster-monitoring-config` under `metricsServer` section
+
+#### Resource Impact
+
+Once metrics-server is enabled, by default 2 replicas of `metrics-server` instances are being deployed in the openshift-monitoring namespace.
+
+As of OpenShift 4.13 the following idle resource usage have been measured:
+
+Kubelet: TBD
+Metrics Server: TBD
+
+**TODO:** Capture the details on performance testing done from [MON-3210](https://issues.redhat.com/browse/MON-3210)
+
+
+### Risks and Mitigations
+
+One of the risk is resource impact on larger clusters. We need to do performance testing on larger OpenShift cluster to measure the impact on Kubelet when metrics-server is deployed.
+
+[MON-3210](https://issues.redhat.com/browse/MON-3210) is created for doing this.
+
+### Drawbacks
+
+N/A
+
+
+## Design Details
+
+### Open Questions [optional]
+
+[Custom Metrics Autoscaler](https://github.com/openshift/custom-metrics-autoscaler-operator)(CMA) also deploys [metrics-server](https://github.com/openshift/custom-metrics-autoscaler-operator/blob/39aa2b9d5a0db2206d4963fe2e551178e7d369ad/controllers/keda/kedacontroller_controller.go#L204) when user creates keda controller after its installed, will this affect Metrics Server we deploy in monitoring stack?
+
+### Test Plan
+
+The proposed solution will be tested with two suites:
+
+* Unit tests - running independently of each other and without side effects
+* End to end tests - in the same repository as the cluster-monitoring-operator. These tests will be executed against an actual Kubernetes cluster and are intended to verify end-to-end functionality of the entire system.
+
+### Graduation Criteria
+
+* 4.14 TechPreview
+  * Notice in the release notes that prometheus-adapter is deprecated (should be repeated on every future release).
+* 4.15  GA as opt-in
+* 4.16 GA as opt-in, warn about prometheus-adapter being deprecated if `prometheus-adapter` config isn’t empty and `metrics-server` isn’t already enabled.
+* 4.17 GA, enabled by default
+* 4.18 => Remove `prometheus-adapter` from CMO
+
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+More testing (upgrade, downgrade, scale).
+* Sufficient time for feedback including signals from telemetry about the customer adoption (e.g. number of metrics-server deployment across the fleet).
+* Counter-measures to avoid service degradation of the Kubelet.
+* Conduct load testing.
+
+#### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+### Upgrade / Downgrade Strategy
+
+TBD
+
+### Version Skew Strategy
+
+TBD
+
+### Operational Aspects of API Extensions
```

Comment:
Please add details about which metrics and alerting rules can be used to monitor metrics-server (even existing alerts.
With telemetry, we can get insights on how well/bad metrics-server works.

Reply:
KubeAggregatedAPIErrors and KubeAggregatedAPIDown already exist today.

Reply:
metrics-server actually has its own metrics on top of the Kubernetes ones: https://github.com/kubernetes-sigs/metrics-server/blob/796fc0f832c1ac444c44f88a952be87524456e07/pkg/server/metrics.go#L29-L46